subsys/net/lib/prometheus/Kconfig

```diff
@@ -0,0 +1,32 @@
+# Copyright(c) 2024 Sparse Technology, Mustafa Abdullah Kus
+# SPDX-License-Identifier: Apache-2.0
+
+menuconfig PROMETHEUS
+	bool "Prometheus Client Library (Pull Method)"
+	depends on NET_SOCKETS
+	depends on HTTP_SERVER
+	depends on HEAP_MEM_POOL_SIZE != 0
```

Comment:
If heap is required, you can create a special Kconfig which specifies how much PROMETHEUS requires, which will be summed up by the build system :)

```
if PROMETHEUS

config HEAP_MEM_POOL_ADD_SIZE_PROMETHEUS
        int "Heap size required by prometheus library in bytes"
        default 2048
        range 256 65535

endif # PROMETHEUS
```

You can then remove `depends on HEAP_MEM_POOL_SIZE != 0` as this becomes explicit

Reply:
I did remove it. I did missed because before force-pushed, the library use memory pool now the metrics create compile time :). Maybe i can add dynamic metric create feature with that config

Reply:
Perfect :)